pkg/kubelet/kubelet_node_status.go

```diff
@@ -148,6 +150,25 @@ func (kl *Kubelet) addManagementNodeCapacity(initialNode, existingNode *v1.Node)
 	return true
 }
 
+func (kl *Kubelet) reconcileSharedCPUsNodeCapacity(initialNode, existingNode *v1.Node) bool {
+	updateDefaultResources(initialNode, existingNode)
+	sharedCPUsResourceName := sharedcpus.GetResourceName()
+	// delete resources in case they exist and feature has been disabled
+	if !sharedcpus.IsEnabled() {
+		if _, ok := existingNode.Status.Capacity[sharedCPUsResourceName]; ok {
+			delete(existingNode.Status.Capacity, sharedCPUsResourceName)
+			return true
+		}
+		return false
+	}
```

Comment:
I like this, though we don't do it for managed cpus. Any reason for the difference in behavior?

Reply:
By managed CPUs you mean the native CPUs?

And the reason for doing that is because after you disable the feature you don't want to  expose those shared CPUs anymore.

Otherwise, workloads might still request for it, get admitted and then crush because somewhere down the line the feature is off.

Reply:
note that we don't delete the resources which are no longer available in general. This is a well known pain point for devices in general. We rather set the capacity to zero and keep the devices lingering.
I think we should keep this consistent (but I also agree much more this behavior, granted we explore the platform/scheduling implications)

Reply:
> I think we should keep this consistent 

The fact that this is not a usual device (it's not populate by device plugin) was my justification to completely remove it and not set as zero.
 from scheduling implication AFAIK zero or lack of a resource should lead to same behavior.

Reply:
Your point is valid indeed.
We can zeroed the resources instead, but let's wait for EP reviewers call as you suggested.